### PR TITLE
Partly fixes #2253: Added NocaseList and corresponding tests

### LINKED ISSUE #2253: Mock: provider registry list methods are case sensitive
ProviderRegistry methods for listing namespaces `provider_namespaces()` and class names `provider_classes()` return case sensitive lists. For example:
```
'Interop' in registry.provider_namespaces()
```
does not properly support a case insensitive test, and thus requires the user to ensure case insensitive comparisons and tests. This can easily be forgotten and should ideally be done automatically, e.g. by returning a case insensitively comparing list.

### PR DESCRIPTION
**Pull request by andy-maier**
This PR adds a `NocaseList` class, but does not use it yet for anything other than testing.

The implementation and test cases are complete now.

### CODE REVIEW

File: pywbem/_nocaselist.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+compared, that is done case-insensitively. Whenever the value of list items
+is returned, they have the lexical case that was originally specified.
+
+It is used for lists of case-insensitive items such as namespaces returned
+by the pywbem mock support.
+
+Users of pywbem will notice ``NocaseList`` objects only as a result of pywbem
+functions. Users cannot create ``NocaseList`` objects.
+
+Except for the case-insensitivity of its items, it behaves like the built-in
```
Comment (KSchopmeyer) [inline pywbem/_nocaselist.py:31]: Is there anything that can be done with a list that cannot be done with the no case list? Ex. iteration/

Comment (andy-maier) [reply pywbem/_nocaselist.py:31]: No.

File: docs/changes.rst [hunk 2]
```diff
@@ -42,6 +42,10 @@ Released: not yet
 
 **Enhancements:**
 
+* Added a new class `pywbem.NocaseList` that is a case-insensitive but case-preserving
+  list. Objects of this class cannot be created by users, but are returned
```
Comment (KSchopmeyer) [inline docs/changes.rst:46]: This means that we have to duplicate this in pywbemtools to get the same functionality or use an internal class from pywbem.

Comment (andy-maier) [reply docs/changes.rst:46]: Should we make it public? And NocaseDict as well?

**Comment by KSchopmeyer** [comment]
APPROVED.

**Comment by andy-maier** [comment]
This PR got manually unapplied from master, so legally, it never went in.

**Pull request merged**